ext/bcmath/libbcmath/src/div.c

```diff
@@ -31,219 +31,436 @@
+	 *   = (numerator_high * divisor_low - divisor_high * numerator_low) / (divisor_high * (divisor_high * B^k + divisor_low))
+	 *
+	 * Find the error MAX_E when the error E is maximum (0 <= E <= MAX_E).
+	 * First, numerator_high, which only exists in the numerator, uses its maximum value.
+	 * Considering carry-back, numerator_high can be expressed as follows.
+	 * numerator_high = divisor_high * B
+	 * Also, numerator_low is only present in the numerator, but since this is a subtraction, use the smallest possible value here, 0.
+	 * numerator_low = 0
+	 *
+	 * MAX_E = (numerator_high * divisor_low - divisor_high * numerator_low) / (divisor_high * (divisor_high * B^k + divisor_low))
+	 *       = (divisor_high * B * divisor_low) / (divisor_high * (divisor_high * B^k + divisor_low))
+	 *
+	 * divisor_low uses the maximum value.
+	 * divisor_low = B^k - 1
+	 * MAX_E = (divisor_high * B * divisor_low) / (divisor_high * (divisor_high * B^k + divisor_low))
+	 *       = (divisor_high * B * (B^k - 1)) / (divisor_high * (divisor_high * B^k + B^k - 1))
+	 *
+	 * divisor_high uses the minimum value, but want to see how the number of digits affects the error, so represent
+	 * the minimum value as:
+	 * divisor_high = 10^x (any x ∈ ℕ)
+	 * Since B = 10^b, the formula becomes:
+	 * MAX_E = (divisor_high * B * (B^k - 1)) / (divisor_high * (divisor_high * B^k + B^k - 1))
+	 *       = (10^x * 10^b * ((10^b)^k - 1)) / (10^x * (10^x * (10^b)^k + (10^b)^k - 1))
+	 *
+	 * Now let's make an approximation. Remove -1 from the numerator. Approximate the numerator to be
+	 * large and the denominator to be small, such that MAX_E is less than this expression.
+	 * Also, the denominator "(10^b)^k - 1" is never a negative value, so delete it.
+	 *
+	 * MAX_E = (10^x * 10^b * ((10^b)^k - 1)) / (10^x * (10^x * (10^b)^k + (10^b)^k - 1))
+	 * MAX_E < (10^x * 10^b * (10^b)^k) / (10^x * 10^x * (10^b)^k)
+	 *       < 10^b / 10^x
+	 *       < 10^(b - x)
+	 *
+	 * Therefore, found that if the number of digits in base B and divisor_high are equal, the error will be less
+	 * than 1 regardless of the number of digits in the value of k.
+	 *
+	 * Here, B is BC_VECTOR_BOUNDARY_NUM, so adjust the number of digits in high part of divisor to be BC_VECTOR_SIZE + 1.
+	 */
+	size_t divisor_top_digits = divisor_len % BC_VECTOR_SIZE;
+	if (divisor_top_digits == 0) {
+		divisor_top_digits = BC_VECTOR_SIZE;
+	}
+	size_t high_part_shift = POW_10_LUT[BC_VECTOR_SIZE - divisor_top_digits + 1];
+	size_t low_part_shift = POW_10_LUT[divisor_top_digits - 1];
+	BC_VECTOR tmp_divisor = divisor_vector[divisor_top_index] * high_part_shift + divisor_vector[divisor_top_index - 1] / low_part_shift;
+	for (size_t i = 0; i < quot_arr_size; i++) {
+		BC_VECTOR tmp_numerator = numerator_vector[numerator_top_index - i] * high_part_shift + numerator_vector[numerator_top_index - i - 1] / low_part_shift;
```

Comment:
```suggestion
		BC_VECTOR numerator_high_part = numerator_vector[numerator_top_index - i] * high_part_shift + numerator_vector[numerator_top_index - i - 1] / low_part_shift;
```

And similar for `tmp_divisor` this would make stuff clearer and remove the need for some comments.

Reply:
Fixed
https://github.com/php/php-src/pull/14660/commits/5b70842c05c317f98df601754200c5d70470d7b1